internal/envtest/environment.go

```diff
@@ -185,13 +186,24 @@ func new(uncachedObjs ...client.Object) *Environment {
 		objs = append(objs, uncachedObjs...)
 	}
 
+	// Localhost is used on MacOS to avoid Firewall warning popups.
+	host := "localhost"
+	if strings.ToLower(os.Getenv("USE_EXISTING_CLUSTER")) == "true" {
+		// 0.0.0.0 is required on Linux when using kind because otherwise the kube-apiserver running in kind
+		// is unable to reach the webhook, because the webhook would be only binded on 127.0.0.1.
+		// Somehow that's not an issue on MacOS.
+		if goruntime.GOOS == "linux" {
+			host = "0.0.0.0"
+		}
+	}
```

Comment:
I'm not sure about this part. We can either keep it as is, or I can make it configurable via env var.

Reply:
Let's keep it as it is and eventually iterate if there are feedback